faunadb-java/src/test/java/com/faunadb/client/JavaClientSpec.java

```diff
@@ -133,14 +122,68 @@ public void shouldGetKeyFromSecret() throws Exception {
       equalTo(rootClient.query(KeyFromSecret(secret)).get()));
   }
 
+  @Test
+  public void shouldTestNestedReferences() throws Exception {
```

Comment:
This test seems duplicated. What's the reason for it?

Reply:
Because `FaunaClient` type doesn't exist on the `java-dsl` project (the common project for both java and android projects). In this test I had to create new instances of `FaunaClient`.